### DISCUSSION TITLE
wasm in Webpack 4

### DISCUSSION THREAD

**Discussion by JoviDeCroock** [webpack/webpack#14042] [Q&A] [answered]
Hey all,

I was trying to get `@swc/wasm-web` to work in a webpack 4 project [astexplorer.net](https://github.com/fkling/astexplorer) but to no avail, I keep running into the following error:

```
ERROR in ./node_modules/@swc/wasm-web/wasm.js 190:42
Module parse failed: Unexpected token (190:42)
You may need an appropriate loader to handle this file type, currently no loaders are configured to process this file. See https://webpack.js.org/concepts#loaders
| async function init(input) {
|   if (typeof input === 'undefined') {
>     input = new URL('wasm_bg.wasm', import.meta.url);
|   }
|   const imports = {};
 @ ./src/parsers/js/swc.js 35:37-69
 @ ./src/parsers sync ^\.\/(?!utils|transpilers)[^/]+\/(transformers\/([^/]+)\/)?(codeExample\.txt|[^/]+?\.js)$
 @ ./src/parsers/index.js
 @ ./src/storage/gist.js
 @ ./src/app.js
```

Is this a lost cause in Webpack 4 as I know the wasm-support has improved in version 5?

I have tried several things like adding loaders for `.wasm`, ... but it feels like webpack is having more issues with the whole rather than just the `.wasm`.

My current try can be found [here](https://github.com/JoviDeCroock/astexplorer/commit/c708ba44af16cd9ef17147569042a451a792f60f)

I've tried the `webassembly/experimental` type, now it has `javascript/auto` with `file-loader`. In the swc repo it seems to [work with v4](https://github.com/swc-project/swc/blob/master/wasm/example/index.js)

Sorry if this is a bit of a long-shot feel free to close if so.

**Comment by alexander-akait** [comment, marked as answer]
webpack v4 doesn't support `import.meta.url`, you need webpack v5

**Reply by alexander-akait** [reply]
Also you don't need `file-loader` for webpack v5 and `new URL(...)` syntax

**Reply by JoviDeCroock** [reply]
So it would be best to go through the effort of upgrading this project to webpack v5, I was hoping to avoid that 😅 but yes that would be a good effort for future-proofing it, thank you _so so so much_ for answering I'm aware this was a long-shot

**Reply by StevenACoffman** [reply]
@JoviDeCroock I made a run at this to get it working here using the draft PR: https://github.com/StevenACoffman/astexplorer/tree/feat/webpack5

Not sure if it helps you anymore, but thought I'd drop a breadcrumb for others.
